blog/_posts/2020-1-23-crash-course-kubernetes-nodes.md

```diff
@@ -0,0 +1,168 @@
+* Compares it to the desired state stored in etcd; and if needed,
+* Calls out to the **API server** to make any changes to the cluster. 
+
+For example, if the Replication Controller sees that the desired state of ReplicaSet A is “four
+copies of podA”, but only three podA’s exist in the cluster, the controller asks the API server
+to spin up another podA.
+
+The **API server** is the bit of master node machinery that makes changes in the cluster. K8s
+internals call to the API server to create, update, and destroy objects, store data in etcd,
+and a bunch of other stuff. The API server also accepts user commands--any `kubectl` command
+you run is a request to the API server.
+
+So when you call `kubectl apply -f pod.yaml`, what's going on under the hood looks sooomething like this:
+
+<!-- Adapted from https://codepen.io/gabrieleromanato/pen/pKrny -->
+<div class="slider" id="main-slider"><!-- outermost container element -->
+	<div class="slider-wrapper"><!-- innermost wrapper element -->
+	    <!-- slides -->
+		<div class="slide" data-image="/assets/images/crash-course-kubernetes-nodes/slide1.png"></div>
+		<div class="slide" data-image="/assets/images/crash-course-kubernetes-nodes/slide2.png"></div>
+		<div class="slide" data-image="/assets/images/crash-course-kubernetes-nodes/slide3.png"></div>
+		<div class="slide" data-image="/assets/images/crash-course-kubernetes-nodes/slide4.png"></div>
+		<div class="slide" data-image="/assets/images/crash-course-kubernetes-nodes/slide5.png"></div>
+		<div class="slide" data-image="/assets/images/crash-course-kubernetes-nodes/slide6.png"></div>
+		<div class="slide" data-image="/assets/images/crash-course-kubernetes-nodes/slide7.png"></div>
+		<div class="slide" data-image="/assets/images/crash-course-kubernetes-nodes/slide8.png"></div>
+		<div class="slide" data-image="/assets/images/crash-course-kubernetes-nodes/slide9.png"></div>
+		<div class="slide" data-image="/assets/images/crash-course-kubernetes-nodes/slide10.png"></div>
+	</div>
+	<div class="slider-nav"><!-- "Previous" and "Next" actions -->
+		<button class="slider-previous">Previous</button>
+		<button class="slider-next">Next</button>
+	</div>
+</div>	
+
+
+To read more about the above components, and all the other bits and pieces on the master node
+that do important work, see [the Kubernetes docs](https://kubernetes.io/docs/concepts/overview/components/#master-components).
+
+### Worker Node(s)
+All the fine-grained control of the master node does no good if there’s nowhere for your apps to
+run. That’s where worker nodes come in. Worker nodes are machines that run one or more pods
+(i.e. units of your app).
+
+If you’re connected to a Kubernetes cluster in the cloud, you’ll most likely run your pods across
+multiple nodes. If you’re running Kubernetes locally (e.g. Docker Desktop, KinD, Minikube), you’ll
+be working with a single node, which is a VM running on your computer. Regardless of the number
```

Comment:
maybe nitty, and I don't know how KinD or Minikube work, but my understanding is that with Docker Desktop on Mac, nodes are containers running inside a VM. I wouldn't be surprised if with KinD or Minikube on Linux, nodes are just containers, with no VM involved. I think I'd be surprised if there were a case where a node was just directly a VM itself.

Reply:
IIRC minikube is a VM and kind is just containers

Reply:
yep, fixed